src/ray/object_manager/plugin_manager.cc

```diff
@@ -0,0 +1,171 @@
+#include "ray/object_manager/plugin_manager.h"
+
+#include "nlohmann/json.hpp"
+
+using json = nlohmann::json;
+
+namespace ray {
+
+using DefaultClientCreator = std::shared_ptr<plasma::ObjectStoreClientInterface> (*)();
+using DefaultRunnerCreator = std::unique_ptr<plasma::ObjectStoreRunnerInterface> (*)(
+                                std::string,
+                                int64_t,
+                                bool,
+                                std::string,
+                                std::string);
+using PluginRunnerCreator = std::unique_ptr<plasma::ObjectStoreRunnerInterface> (*)();
+using PluginClientCreator = std::shared_ptr<plasma::ObjectStoreClientInterface> (*)();
+
+PluginManager &PluginManager::GetInstance() {
+  static PluginManager instance;
+  return instance;
+}
+
+/// A creator function of default object store client (PlasmaClient).
+std::shared_ptr<plasma::ObjectStoreClientInterface> CreateDefaultClientInstance() {
+  return std::make_shared<plasma::PlasmaClient>();
+}
+
+/// A creator function of default object store runner (PlasmaStoreRunner).
+std::unique_ptr<plasma::ObjectStoreRunnerInterface> CreateDefaultRunnerInstance(
+    std::string store_socket_name,
+    int64_t object_store_memory,
+    bool huge_pages,
+    std::string plasma_directory,
+    std::string fallback_directory) {
+  return std::make_unique<plasma::PlasmaStoreRunner>(store_socket_name,
+                                                     object_store_memory,
+                                                     huge_pages,
+                                                     plasma_directory,
+                                                     fallback_directory);
+}
+
+void PluginManager::LoadObjectStorePlugin() {
+  void *handle = dlopen(plugin_path_.c_str(), RTLD_NOW);
+  if (!handle) {
+    std::cerr << "Failed to load shared library: " << dlerror() << std::endl;
+    return;
+  }
+
+  PluginClientCreator client_creator =
+      reinterpret_cast<PluginClientCreator>(dlsym(handle, "CreateNewStoreClient"));
```

Comment:
Are "Create" and "New" kind of redundant?

Reply:
I will change the name in the example plugin. How about `CreateExternalStoreClient` ?

Reply:
How about CreateObjectStoreClient, or just CreateClient?